public/media_engagement/imaqal_media_engagement.js

```diff
@@ -0,0 +1,455 @@
+//Permorm Authentication then update data 
+firebase.auth().onAuthStateChanged(function (user) {
+    if (user) {
+        console.log("Attempting to bind: " + user.email)
+        const mediadb = firebase.firestore();
+        const settings = { timestampsInSnapshots: true };
+        mediadb.settings(settings);
+        var data = [];
+        mediadb.collection('/metrics/rapid_pro/IMAQAL/').onSnapshot(res => {
+            console.log(res)
+            // Update data every time it changes in firestore
+            res.docChanges().forEach(change => {
+
+                const doc = { ...change.doc.data(), id: change.doc.id };
+
+                switch (change.type) {
+                    case 'added':
+                        data.push(doc);
+                        break;
+                    case 'modified':
+                        const index = data.findIndex(item => item.id == doc.id);
+                        data[index] = doc;
+                        break;
+                    case 'removed':
+                        data = data.filter(item => item.id !== doc.id);
+                        break;
+                    default:
+                        break;
+                }
+            });
+            update(data);
+        });
+        console.log('Bind Successful');
+    } else {
+        window.location.replace('auth.html')
+    }
+});
+
+//Create margins for the two graphs
+const Margin = { top: 40, right: 100, bottom: 50, left: 70 };
+const Width = 900 - Margin.right - Margin.left;
+const Height = 500 - Margin.top - Margin.bottom;
+
+// Append total received sms graph to svg
+var total_received_sms_graph = d3.select(".total_received_sms_graph").append("svg")
+    .attr("width", Width + Margin.left + Margin.right)
+    .attr("height", Height + Margin.top + Margin.bottom)
+    .append("g")
+    .attr("transform",
+        "translate(" + Margin.left + "," + Margin.top + ")");
+
+// Append total sent sms graph to svg
+var total_sent_sms_graph = d3.select(".total_sent_sms_graph").append("svg")
+    .attr("width", Width + Margin.left + Margin.right)
+    .attr("height", Height + Margin.top + Margin.bottom)
+    .append("g")
+    .attr("transform",
+        "translate(" + Margin.left + "," + Margin.top + ")");
+
+// Append total sent sms graph to svg
+var total_failed_sms_graph = d3.select(".total_failed_sms_graph").append("svg")
+    .attr("width", Width + Margin.left + Margin.right)
+    .attr("height", Height + Margin.top + Margin.bottom)
+    .append("g")
+    .attr("transform",
+        "translate(" + Margin.left + "," + Margin.top + ")");
+
+// Format TimeStamp  
+var timeFormat = d3.timeFormat("%d %H:%M:%S");
```

Comment:
Please format as day/month/year

You can do this in a following PR if you wish

Reply:
Updated to hour/day/month/year